ntbs-service/Services/AuthorizationService.cs

```diff
@@ -58,18 +58,21 @@ async Task SetPadlockForBannerAsync(ClaimsPrincipal u, NotificationBannerModel b
                 _userPermissionsFilter = await GetUserPermissionsFilterAsync(user);
             }
 
-            if (notification.NotificationStatus == NotificationStatus.Closed &&
-                _userPermissionsFilter.Type != UserType.NationalTeam)
+            if ((UserHasDirectRelationToNotification(notification) && notification.NotificationStatus != NotificationStatus.Closed) 
+                || _userPermissionsFilter.Type == UserType.NationalTeam)
             {
-                return PermissionLevel.ReadOnly;
+                return PermissionLevel.Edit;
             }
-            
-            if (UserHasDirectRelationToNotification(notification) || _userPermissionsFilter.Type == UserType.NationalTeam)
+
+            if (UserHasDirectRelationToNotification(notification) &&
+                notification.NotificationStatus == NotificationStatus.Closed)
             {
-                return PermissionLevel.Edit;
+                return PermissionLevel.ReadOnly;
             }

```

Comment:
Hmm, I wonder if this would be a tad clearer if nested? so something like 
```
            if (UserHasDirectRelationToNotification(notification) 
            {
                return notification.NotificationStatus != NotificationStatus.Closed || _userPermissionsFilter.Type == UserType.NationalTeam
                    ? PermissionLevel.Edit
                    : PermissionLevel.ReadOnly;
            }
```

Alternatively, rather then `return`ing straight away we could play around with having a variable that we set and if the notification is closed we "downgrade" the permission level from edit to readonly? Not so sure about this one tbh, but thought it might be worth throwing it out there.

Reply:
Yeah I was tinkering with the first option before separating it out, now that I'm seeing it suggested again I'm leaning towards it, think that's the best solution!